packages/admin/admin/src/dataGrid/GridCellText.tsx

```diff
@@ -0,0 +1,107 @@
+import { ComponentsOverrides, css, Theme, Typography, useThemeProps } from "@mui/material";
+import React from "react";
+
+import { createComponentSlot } from "../helpers/createComponentSlot";
+import { ThemedComponentBaseProps } from "../helpers/ThemedComponentBaseProps";
+
+export type GridCellTextClassKey = "root" | "hasSecondaryText" | "primaryText" | "secondaryText";
+
+export interface GridCellTextProps
+    extends ThemedComponentBaseProps<{
+        root: "div";
+        primaryText: typeof Typography;
+        secondaryText: typeof Typography;
+    }> {
+    primary?: React.ReactNode;
+    secondary?: React.ReactNode;
+    children?: React.ReactNode;
+}
+
+type OwnerState = {
+    hasSecondaryText: boolean;
+};
+
+export const GridCellText = (inProps: GridCellTextProps) => {
+    const { children, primary, secondary, slotProps, ...restProps } = useThemeProps({ props: inProps, name: "CometAdminGridCellText" });
+
+    const ownerState: OwnerState = {
+        hasSecondaryText: Boolean(secondary),
+    };
+
+    return (
+        <Root ownerState={ownerState} {...slotProps?.root} {...restProps}>
+            <PrimaryText ownerState={ownerState} {...slotProps?.primaryText}>
+                {children ? children : primary}
```

Comment:
Do you think we should throw an error when both or none of those props are provided?